src/pages/Admin/Charity/Permissions/Table.tsx

```diff
@@ -0,0 +1,141 @@
+            control={control}
+            name={formField}
+            render={({ field: { value, onChange } }) => {
+              const isDisabled =
+                !value.modifiable ||
+                disabled ||
+                (initialValues[formField].delegate &&
+                  !!initialValues[formField].delegate_address &&
+                  initialValues[formField].delegate_address !==
+                    wallet?.address);
+              return (
+                <Cells
+                  type="td"
+                  cellClass="py-3 px-4 border-t border-prim min-w-[116px] max-w-xs truncate first:rounded-bl last:rounded-br"
+                >
+                  <>{value.name}</>
+
+                  <input
+                    type="checkbox"
+                    className="checkbox-orange"
+                    checked={value.owner_controlled}
+                    disabled={isDisabled}
+                    onChange={() =>
+                      onChange({
+                        ...value,
+                        owner_controlled: !value.owner_controlled,
+                      })
+                    }
+                  />
+                  {isNormal ? (
+                    <input
+                      type="checkbox"
+                      className="checkbox-orange"
+                      checked={value.gov_controlled}
+                      disabled={isDisabled}
+                      onChange={() =>
+                        onChange({
+                          ...value,
+                          gov_controlled: !value.gov_controlled,
+                        })
+                      }
+                    />
+                  ) : null}
+                  <input
+                    type="checkbox"
+                    className="checkbox-orange"
+                    checked={value.delegate}
+                    disabled={isDisabled}
+                    onChange={() =>
+                      onChange({
+                        ...value,
+                        delegate: !value.delegate,
+                      })
+                    }
+                  />
+
+                  <input
+                    type="text"
+                    className="field-input w-full truncate py-1.5"
+                    placeholder="Wallet address..."
+                    value={value.delegate_address}
+                    onChange={(e) =>
+                      onChange({
+                        ...value,
+                        delegate_address: e.target.value,
+                      })
+                    }
+                    disabled={!value.delegate || isDisabled}
+                  />
+
+                  {isNormal ? (
+                    <button
+                      type="button"
+                      className="btn-red py-1 px-2 rounded font-semibold text-xs uppercase text-white tracking-wider"
+                      disabled={isDisabled}
+                      onClick={() => {
```

Comment:
<img width="1165" alt="image" src="https://user-images.githubusercontent.com/89639563/225187912-c074e360-531d-480d-a963-22bf779dc677.png">

`lock-forever` should trigger validation, we could get this scenario that an invalid value is locked

Reply:
@SovereignAndrey @ap-justin 
Maybe we can make it so that clicking "Lock forever" just sets the appropriate form value, but does not **lock** the table row until the form is submitted and the changes are executed.
In other words, make it so that a table row is _actually locked_ only if the initial `modifiable` field is set to `false`. This would allow for two things:
- no need to manually trigger validation
- **even more important** - would allow the user to go back and forth between "Lock forever" and "LockED forever", which they would be able to do until they actually submit the form (current UX is bad as once the settings are set as "locked" but NOT SUBMITTED, they cannot be switched back until the whole page is reloaded - what if the user locked the settings by mistake or something dumb as just wanting to see what would happen). In this case, once the form is submitted and executed, the rows are locked *for real*.

Reply:
> @SovereignAndrey @ap-justin Maybe we can make it so that clicking "Lock forever" just sets the appropriate form value, but does not **lock** the table row until the form is submitted and the changes are executed. In other words, make it so that a table row is _actually locked_ only if the initial `modifiable` field is set to `false`. This would allow for two things:
> 
> * the user would be able to keep updating other fields in the same row -> just because they want the fields locked in the future, does not mean they want them locked _immediately_ - they might still want to change other fields in the same row and THEN lock them (once submitted)
> * would allow the user to go back and forth between "Lock forever" and "LockED forever", which they would be able to do until they actually submit the form (current UX is bad as once the settings are set as "locked" but NOT SUBMITTED, they cannot be switched back until the whole page is reloaded - what if the user locked the settings by mistake or something dumb as just wanting to see what would happen). In this case, once the form is submitted and executed, the rows are locked _for real_.

Agreed, that we should allow them to configure the settings (owner and locked status (assuming still modifiable)) to however they wish before submitting and send everything at once. That said locking a previously unlocked field should "grey out" the row or something to signal it's all set in stone at that point, but they should be able to toggle that back to "unlocked" and change the ownership checks / delegate fields if desired. 

We'll want to make sure that if a field is originally locked on contract then it can NOT be toggled back to unlocked. Allowing that would mean a submitted message / proposal would fail since it'd be trying to update a non-modifiable field in the EndowmentController. 

* I will need to update the contract's `set_permissions` function to check the existing `modifiable` state there _before_ setting new values. We currently only look at modifiable when it comes to changing the field's setting and not the controller as well.